go/vt/vtgate/executor.go

```diff
@@ -231,7 +234,7 @@ func (e *Executor) Execute(ctx context.Context, mysqlCtx vtgateservice.MySQLConn
 		if err != nil {
 			piiSafeSQL = logStats.StmtType
 		}
-		log.Warningf("%q exceeds warning threshold of max memory rows: %v. Actual memory rows: %v", piiSafeSQL, warnMemoryRows, len(result.Rows))
+		exceedMemoryRowsLogger.Warningf("%q exceeds warning threshold of max memory rows: %v. Actual memory rows: %v", piiSafeSQL, warnMemoryRows, len(result.Rows))
```

Comment:
Does this throttle only on the exact matching `len(result.Rows)` or also on different lengths? And should it only throttle on exactly the same or should we also throttle if it's a slightly different value?

Reply:
It feels like the throttler will throttle all max memory warning regardless of the query. I think it would be more useful if we could throttle based on the query we are receiving rather than just the error.

Reply:
Right now it will throttle all log messages that are generated by this line. That is, is might be a different query, or it might be a different number of rows. It will only print one message a minute.
I agree that it will be nice to print one message per distinct query. That will need writing a new kind of ThrottledLogger. I don't know if that is worth it. If we believe it is, then let's do it.